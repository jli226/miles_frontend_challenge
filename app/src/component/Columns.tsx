import React from "react";
import { useStore } from "../store/useStore";
import { observer } from "mobx-react";
import Column from "./Column";

const Columns = observer(() => {
  const store = useStore();
  const columns = store.categories.map((category) => (
    <Column key={category} category={category} />
  ));
  return <div className="columns">{columns}</div>;
});

export default Columns;
