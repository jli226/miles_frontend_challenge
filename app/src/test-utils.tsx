import React from "react";
import "mobx-react/batchingForReactDom";
import { render } from "@testing-library/react";
import { StoreProvider } from "./store/useStore";
import { DndProvider } from "react-dnd";
import { TestBackend } from "react-dnd-test-backend";

export const renderWithProviders = (children: React.ReactChild) =>
  render(
    <StoreProvider>
      <DndProvider backend={TestBackend}>{children}</DndProvider>
    </StoreProvider>
  );
