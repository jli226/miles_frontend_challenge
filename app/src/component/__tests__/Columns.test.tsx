import React from "react";
import { renderWithProviders } from "../../test-utils";
import Columns from "../Columns";

test("renders the Column in a row", () => {
  const { container } = renderWithProviders(<Columns />);
  expect(container).toMatchSnapshot();
});
