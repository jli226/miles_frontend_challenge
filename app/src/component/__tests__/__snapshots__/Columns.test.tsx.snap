// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Column in a row 1`] = `
<div>
  <div
    class="columns"
  >
    <div
      class="lane dropzone"
    >
      <div
        class="header"
      >
        C1
      </div>
      <div
        class="indicators"
      >
        <div
          class="indicator"
        >
          <div
            class="close"
          />
          R1
        </div>
        <div
          class="indicator"
        >
          <div
            class="close"
          />
          R2
        </div>
        <div
          class="indicator"
        >
          <div
            class="close"
          />
          R3
        </div>
        <div
          class="indicator"
        >
          <div
            class="close"
          />
          R4
        </div>
        <div
          class="indicator"
        >
          <div
            class="close"
          />
          R5
        </div>
      </div>
    </div>
    <div
      class="lane dropzone"
    >
      <div
        class="header"
      >
        C2
      </div>
      <div
        class="indicators"
      >
        <div
          class="indicator"
        >
          <div
            class="close"
          />
          R1
        </div>
        <div
          class="indicator"
        >
          <div
            class="close"
          />
          R2
        </div>
        <div
          class="indicator"
        >
          <div
            class="close"
          />
          R3
        </div>
        <div
          class="indicator"
        >
          <div
            class="close"
          />
          R4
        </div>
        <div
          class="indicator"
        >
          <div
            class="close"
          />
          R5
        </div>
      </div>
    </div>
    <div
      class="lane dropzone"
    >
      <div
        class="header"
      >
        C3
      </div>
      <div
        class="indicators"
      >
        <div
          class="indicator"
        >
          <div
            class="close"
          />
          R1
        </div>
        <div
          class="indicator"
        >
          <div
            class="close"
          />
          R2
        </div>
        <div
          class="indicator"
        >
          <div
            class="close"
          />
          R3
        </div>
        <div
          class="indicator"
        >
          <div
            class="close"
          />
          R4
        </div>
        <div
          class="indicator"
        >
          <div
            class="close"
          />
          R5
        </div>
      </div>
    </div>
    <div
      class="lane dropzone"
    >
      <div
        class="header"
      >
        C4
      </div>
      <div
        class="indicators"
      >
        <div
          class="indicator"
        >
          <div
            class="close"
          />
          R1
        </div>
        <div
          class="indicator"
        >
          <div
            class="close"
          />
          R2
        </div>
        <div
          class="indicator"
        >
          <div
            class="close"
          />
          R3
        </div>
        <div
          class="indicator"
        >
          <div
            class="close"
          />
          R4
        </div>
        <div
          class="indicator"
        >
          <div
            class="close"
          />
          R5
        </div>
      </div>
    </div>
    <div
      class="lane dropzone"
    >
      <div
        class="header"
      >
        C5
      </div>
      <div
        class="indicators"
      >
        <div
          class="indicator"
        >
          <div
            class="close"
          />
          R1
        </div>
        <div
          class="indicator"
        >
          <div
            class="close"
          />
          R2
        </div>
        <div
          class="indicator"
        >
          <div
            class="close"
          />
          R3
        </div>
        <div
          class="indicator"
        >
          <div
            class="close"
          />
          R4
        </div>
        <div
          class="indicator"
        >
          <div
            class="close"
          />
          R5
        </div>
      </div>
    </div>
  </div>
</div>
`;
